sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaSourceDescription.java

```diff
@@ -0,0 +1,92 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.kafka;
+
+import com.google.auto.value.AutoValue;
+import java.io.Serializable;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.NoSuchSchemaException;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.SchemaCoder;
+import org.apache.beam.sdk.schemas.SchemaRegistry;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.sdk.schemas.annotations.SchemaFieldName;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.kafka.common.TopicPartition;
+import org.joda.time.Instant;
+
+/**
+ * An AutoValue object which represents a Kafka source description. Note that this object should be
+ * encoded/decoded with equivalent {@link Schema} as a {@link Row} when crossing the wire.
+ */
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class KafkaSourceDescription implements Serializable {
+  @SchemaFieldName("topic")
+  abstract String getTopic();
+
+  @SchemaFieldName("partition")
+  abstract Integer getPartition();
+
+  @SchemaFieldName("start_read_offset")
+  @Nullable
+  abstract Long getStartReadOffset();
+
+  @SchemaFieldName("start_read_time")
+  @Nullable
+  abstract Instant getStartReadTime();
+
+  @SchemaFieldName("bootstrapServers")
+  @Nullable
+  abstract List<String> getBootStrapServers();
+
+  private TopicPartition topicPartition = null;
+
+  public TopicPartition getTopicPartition() {
+    if (topicPartition == null) {
+      topicPartition = new TopicPartition(getTopic(), getPartition());
+    }
+    return topicPartition;
+  }
+
+  public static KafkaSourceDescription of(
+      TopicPartition topicPartition,
+      Long startReadOffset,
+      Instant startReadTime,
+      List<String> bootstrapServers) {
+    return new AutoValue_KafkaSourceDescription(
+        topicPartition.topic(),
+        topicPartition.partition(),
+        startReadOffset,
+        startReadTime,
+        bootstrapServers);
+  }
+
+  public static Coder<KafkaSourceDescription> getCoder(SchemaRegistry registry)
+      throws NoSuchSchemaException {
+    return SchemaCoder.of(
+        registry.getSchema(KafkaSourceDescription.class),
+        TypeDescriptor.of(KafkaSourceDescription.class),
+        registry.getToRowFunction(KafkaSourceDescription.class),
+        registry.getFromRowFunction(KafkaSourceDescription.class));
+  }
```

Comment:
I don't think you you should need this function. By registering a schema for KafkaSourceDescription Beam we will default to using SchemaCoder